Gems/EMotionFX/Code/EMotionFX/Source/Mesh.cpp

```diff
@@ -769,37 +756,33 @@ namespace EMotionFX
         }
 
         // return the maximum number of influences
-        return aznumeric_cast<uint32>(maxInfluences);
+        return maxInfluences;
     }
 
 
     // returns the maximum number of weights/influences for this mesh plus some extra information
-    uint32 Mesh::CalcMaxNumInfluences(AZStd::vector<uint32>& outVertexCounts) const
+    size_t Mesh::CalcMaxNumInfluences(AZStd::vector<size_t>& outVertexCounts) const
     {
-        size_t maxInfluences = 0;
-
         // Reset values.
         outVertexCounts.resize(CalcMaxNumInfluences() + 1);
-        for (size_t j = 0; j < outVertexCounts.size(); ++j)
-        {
-            outVertexCounts[j] = 0;
-        }
+        AZStd::fill(begin(outVertexCounts), end(outVertexCounts), 0);
 
         // Does the mesh have a skinning layer? If no we can quit directly as this means there are only unskinned vertices.
         SkinningInfoVertexAttributeLayer* skinningLayer = (SkinningInfoVertexAttributeLayer*)FindSharedVertexAttributeLayer(SkinningInfoVertexAttributeLayer::TYPE_ID);
         if (!skinningLayer)
         {
             outVertexCounts[0] = GetNumVertices();
-            return aznumeric_cast<uint32>(maxInfluences);
+            return 0;
```

Comment:
Why are we just returning 0 here while we returned the maxInfluences before?

Reply:
The old code was this:

```cpp
  1 uint32 Mesh::CalcMaxNumInfluences(AZStd::vector<uint32>& outVertexCounts) const
  2 {
  3     size_t maxInfluences = 0;
  4
  5     // Reset values.
  6     outVertexCounts.resize(CalcMaxNumInfluences() + 1);
  7     for (size_t j = 0; j < outVertexCounts.size(); ++j)
  8     {
  9         outVertexCounts[j] = 0;
 10     }
 11
 12     // Does the mesh have a skinning layer? If no we can quit directly as this means there are only unskinned vertic    es.
 13     SkinningInfoVertexAttributeLayer* skinningLayer = (SkinningInfoVertexAttributeLayer*)FindSharedVertexAttributeLa    yer(SkinningInfoVertexAttributeLayer::TYPE_ID);
 14     if (!skinningLayer)
 15     {
 16         outVertexCounts[0] = GetNumVertices();
 17         return aznumeric_cast<uint32>(maxInfluences);
 18     }
```

You can see on line 3 that `maxInfluences` is initialized to `0`, and never modified before the return on line 17. So if there's no skinning layer, it will return `0`. This seems correct to me.